drivers/char/rust_example.rs

```diff
@@ -49,8 +59,13 @@ impl KernelModule for RustExample {
         println!("Rust Example (init)");
         println!("Am I built-in? {}", !cfg!(MODULE));
         println!("Parameters:");
-        println!("  my_bool:  {}", my_bool.read());
-        println!("  my_i32:   {}", my_i32.read());
+        println!("  my_bool:    {}", my_bool.read());
+        println!("  my_i32:     {}", my_i32.read());
+        println!(
+            "  my_str:     {}",
+            my_str.read().expect("Expected valid UTF8 parameter")
```

Comment:
I can see the value on building a proper `str` and supporting UTF-8 right away. On the other hand, I am not sure what people will think about validating/enforcing Unicode, due to C not requiring it (it is treated more like a byte string, from what I can see).

Reply:
By the way, an `expect()` in a driver, particularly for something from userspace looks like a bad idea -- you can kill the entire system by passing a bad parameter (even if only root can install a module and it is just an honest mistake). In the kernel, panicking is usually avoided unless there is a good reason to kill the machine. Even a hanging driver doesn't typically trigger a panic.

Reply:
Yeah I wasn't really sure what people will be more comfortable with here - I guess we can do `&[u8]` and let users decide what they want to do.

Also I'm wondering now if we should  allocate a new `Vec` and copy the bytes over because the original string will be freed if the parameter is changed.

Reply:
I think `&[u8]` is better.

Reply:
> Also I'm wondering now if we should allocate a new `Vec` and copy the bytes over because the original string will be freed if the parameter is changed.

If the parameter is writable, then yeah, we need to copy it or return a lock.

Reply:
Ok, using `&[u8]` now and added locking. But the locking is only available when `CONFIG_SYSFS` is enabled because the locking function `kernel_param_lock` from C isn't defined otherwise.